source/server/hot_restarting_child.cc

```diff
@@ -48,11 +50,42 @@ HotRestartingChild::HotRestartingChild(int base_id, int restart_epoch,
                                        const std::string& socket_path, mode_t socket_mode)
     : HotRestartingBase(base_id), restart_epoch_(restart_epoch) {
   main_rpc_stream_.initDomainSocketAddress(&parent_address_);
+  std::string socket_path_udp = socket_path + "_udp";
+  udp_forwarding_rpc_stream_.initDomainSocketAddress(&parent_address_udp_forwarding_);
   if (restart_epoch_ != 0) {
     parent_address_ = main_rpc_stream_.createDomainSocketAddress(restart_epoch_ + -1, "parent",
                                                                  socket_path, socket_mode);
+    parent_address_udp_forwarding_ = udp_forwarding_rpc_stream_.createDomainSocketAddress(
+        restart_epoch_ + -1, "parent", socket_path_udp, socket_mode);
   }
   main_rpc_stream_.bindDomainSocket(restart_epoch_, "child", socket_path, socket_mode);
+  udp_forwarding_rpc_stream_.bindDomainSocket(restart_epoch_, "child", socket_path_udp,
+                                              socket_mode);
+}
+
+// Destructor must be specified in the cc file because UdpForwardingContext must be defined first
```

Comment:
That seems surprising. Is it a compiler-error if it's located i the normal place in the .h?

Reply:
Ah, this was a bit of a holdover from when UdpForwardingContext was defined in the cc file scope rather than in the header, but it was *also* true where it was. I have rearranged it so it's defined in the header before the destructor, and now the destructor is also =default in the header.